bin/internal/update_dart_sdk.sh

```diff
@@ -71,7 +71,7 @@ if [ ! -f "$ENGINE_STAMP" ] || [ "$ENGINE_VERSION" != `cat "$ENGINE_STAMP"` ]; t
 
   case "$(uname -s)" in
```

Comment:
Let me see if this works on macOS, if the script is running with Rosetta then `uname -m` will return `x86_64`, not `arm64`.  It would have to be detected with `sysctl hw.optional.arm64`:
https://github.com/flutter/flutter/blob/610b41e88cc1f29c8bc9faed0e4d328b26f05b9b/packages/flutter_tools/lib/src/base/os.dart#L422-L426

Reply:
Oh nice, at this point in the flow the script isn't run with Rosetta (which would happen if it was spawned by a Rosetta process), when I `echo` `ARCH` I see `arm64`, the bash process is running it "Apple" which means arm.